docs/root/start/sandboxes/single-page-app.rst

```diff
@@ -0,0 +1,497 @@
+    single-page-app-myhub-1       single-page-app-myhub       "/opt/myhub/app.py"                                        myhub     2 minutes ago   Up 2 minutes   0.0.0.0:7000->7000/tcp, :::7000->7000/tcp
+    single-page-app-myhub-api-1   single-page-app-myhub-api   "/opt/myhub/app.py"                                        myhub-api 2 minutes ago   Up 2 minutes
+    single-page-app-ui-1          single-page-app-ui          "/entrypoint.sh dev.sh"                                    ui        2 minutes ago   Up 2 minutes
+
+.. _install_sandboxes_single_page_app_step_login:
+
+Step 3: Browse to the dev app and login
+***************************************
+
+The development app should now be available on at http://localhost:10001 and provide a login button.
+
+.. image:: /start/sandboxes/_static/spa-login.png
+   :align: center
+
+.. note::
+   The dummy OAuth provider automatically trusts everyone as a hard-coded ``envoydemo`` user and redirects back to the app.
+
+   In a real world scenario the provider would authenticate and authorize the user before proceeding.
+
+When a user clicks ``login`` it initiates the OAuth flow by calling the ``/authorize`` path:
+
+.. literalinclude:: _include/single-page-app/envoy.yml
+   :language: yaml
+   :lines: 44-50
+   :linenos:
+   :lineno-start: 44
+   :emphasize-lines: 3-5
+   :caption: :download:`envoy.yml <_include/single-page-app/envoy.yml>`
+
+This redirects the user to the OAuth provider with a further redirect link.
+
+.. literalinclude:: _include/single-page-app/envoy.yml
+   :language: yaml
+   :lines: 36-41
+   :linenos:
+   :lineno-start: 36
+   :emphasize-lines: 3-4
+   :caption: :download:`envoy.yml <_include/single-page-app/envoy.yml>`
+
+On successful authorization/authentication the user is redirected back to the app with the necessary OAuth code to proceed.
+
+Envoy then uses this code with its client secret to generate an access token for the user which Envoy retains.
+
+.. literalinclude:: _include/single-page-app/envoy.yml
+   :language: yaml
+   :lines: 50-60
+   :linenos:
+   :lineno-start: 50
+   :emphasize-lines: 3-10
+   :caption: :download:`envoy.yml <_include/single-page-app/envoy.yml>`
+
+.. literalinclude:: _include/single-page-app/secrets/myhub-token-secret.yml
+   :language: yaml
+   :linenos:
+   :emphasize-lines: 6
+   :caption: :download:`myhub-token-secret.yml <_include/single-page-app/secrets/myhub-token-secret.yml>`
+
+Once logged in, you should be able to make queries to the API using the OAuth credentials.
+
+.. image:: /start/sandboxes/_static/spa-resources.png
+   :align: center
+
+.. warning::
+   Envoy's OAuth implementation defaults to triggering the OAuth flow for all requests on the endpoint.
+
+   This can readily trigger an OAuth flood as assets are requested, and doom loops when the OAuth flows fail.
+
+   This can be avoided by restricting the paths that are used by the OAuth flow.
```

Comment:
@mattklein123 flagging this - in initial dev i was constantly triggering a browser/server doom loop

i thought perhaps it was my implementation (to some extent this was true), but the more i debugged i realized its kinda endemic if the endpoint assumes all paths should trigger auth flow - relatedly there is some defensive code envoy-side to exit if it detects this (_mostly_ it didnt in my testing)

either way, default usage readily triggers this, and after quite a bit of testing/dev ive found it hard to prevent - not sure current implementation of react app/envoy config would be entirely reliable tbh - not sure how to prevent but i fwiw i do know how to trigger i think

Reply:
a bit more confident in the implementation here now - it inverts pass_through_matcher to only trigger oauth flow for the oauth endpoints - ill update the docs here - possibly we should add something to the filter docs